### PR TITLE
Update server/simple documentation

### PR DESCRIPTION
**Pull request by kevmoo**
Update readmes
Improve inline documentation in bin/server.dart
Cleanup to inedx.html and tests

### CODE REVIEW

**Comment by kevmoo** [comment]
Contributes to https://github.com/dart-lang/site-www/issues/3058

File: server/simple/README.md [hunk 1]
```diff
@@ -3,8 +3,9 @@ A simple Dart HTTP server using [package:shelf](https://pub.dev/packages/shelf).
...
-  the `PORT` environment variable.
+  the `PORT` environment variable. (This is also the convention used by
+  [Google Cloud Run](https://cloud.google.com/run).)
 - Includes `Dockerfile` for easy containerization
-- Follow instructions at
-  https://cloud.google.com/run/docs/quickstarts/build-and-deploy to build and
-  deploy on [Google Cloud Run](https://cloud.google.com/run).
+
+To deploy on [Google Cloud Run](https://cloud.google.com/run), follow
+[these instructions](https://cloud.google.com/run/docs/quickstarts/build-and-deploy/other).
```
Comment (kwalrath) [inline server/simple/README.md:11]: Good clause reordering!

**Review by kwalrath — APPROVED** [review]
LGTM, but I think that Google Cloud Run should just be Cloud Run. (See https://cloud.google.com/products/ for all the official names.)

File: server/simple/README.md (hunk 1 above)
Comment (kevmoo) [reply server/simple/README.md:11]: 🙇

**Comment by kevmoo** [comment]
> LGTM, but I think that Google Cloud Run should just be Cloud Run. (See https://cloud.google.com/products/ for all the official names.)

ack!

**Comment by kevmoo** [comment]
CC @mit-mit FYI

**Pull request merged**